sycl/include/CL/sycl/range.hpp

```diff
@@ -143,6 +165,51 @@ template <int dimensions = 1> class range : public detail::array<dimensions> {
 
 #undef __SYCL_GEN_OPT
 
+// OP is unary +, -
+#define __SYCL_GEN_OPT(op)                                                     \
+  friend range<dimensions> operator op(const range<dimensions> &rhs) {         \
+    range<dimensions> result(rhs);                                             \
+    for (int i = 0; i < dimensions; ++i) {                                     \
+      result.common_array[i] = (op rhs.common_array[i]);                       \
+    }                                                                          \
+    return result;                                                             \
+  }
+
+  __SYCL_GEN_OPT(+)
+  __SYCL_GEN_OPT(-)
+
+#undef __SYCL_GEN_OPT
+
+// OP is prefix ++, --
+#define __SYCL_GEN_OPT(op)                                                     \
+  friend range<dimensions> &operator op(range<dimensions> &rhs) {              \
+    for (int i = 0; i < dimensions; ++i) {                                     \
+      op rhs.common_array[i];                                                  \
+    }                                                                          \
+    return rhs;                                                                \
+  }
+
+  __SYCL_GEN_OPT(++)
+  __SYCL_GEN_OPT(--)
+
+#undef __SYCL_GEN_OPT
+
+// OP is postfix ++, --
+#define __SYCL_GEN_OPT(op)                                                     \
+  friend range<dimensions> operator op(range<dimensions> &lhs, int) {          \
```

Comment:
Why the first argument is called `lhs` while in other operators it's called `rhs`?

Reply:
Because the operator parameters are defined like that in the specification: https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#range-class
```
// OP is postfix ++, --
  friend range operatorOP(range &lhs, int) { /* ... */ }
```